pkg/util/pullsecret/pullsecret.go

```diff
@@ -0,0 +1,30 @@
+package pullsecret
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+)
+
+type RegistryAuth struct {
+	Auth string `json:"auth,omitempty"`
+}
+
+type PullSecret struct {
+	Auths map[string]RegistryAuth `json:"auths,omitempty"`
+}
+
+func ReplaceRegistryAuth(original, registry, username, password string) (string, error) {
+	var pr *PullSecret
+	err := json.Unmarshal([]byte(original), pr)
```

Comment:
consider if []byte is neater

Reply:
you have to convert is somewhere as the os.Getenv() gives you a string